code-challenges.test.js

```diff
@@ -12,35 +12,164 @@
+
 // a) Create a test with an expect statement using the variable provided.
 
-const hitchhikersCharacters = [
-  { name: "ford prefect", occupation: "a hitchhiker" },
-  { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
-  { name: "arthur dent", occupation: "a radio employee" }
-]
-// Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+describe("nameToCap", () => {
+  it("takes in an array of objects and returns an array with a sentence about each person with their name capitalized", () =>{
+    expect(nameToCap(hitchhikersCharacters)).toStrictEqual(["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."])
+    const hitchhikersCharacters = [
+      { name: "ford prefect", occupation: "a hitchhiker" },
+      { name: "zaphod beeblebrox", occupation: "president of the galaxy" },
+      { name: "arthur dent", occupation: "a radio employee" }
+    ]
+    // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is president of the galaxy.", "Arthur Dent is a radio employee."]
+  })
+})
+
+// FAIL  ./code-challenges.test.js
+//   nameToCap
+//     ✕ takes in an array of objects and returns an array with a sentence about each person with their name capitalized (1 ms)
+
+//   ● nameToCap › takes in an array of objects and returns an array with a sentence about each person with their name capitalized
+
+//     ReferenceError: nameToCap is not defined
 
 // b) Create the function that makes the test pass.
 
+// Function nameToCap that takes an array as a parameter 
+const nameToCap = (array)  =>{
+  // Return a new array by mapping each object in the input array
+  return array.map(object => {
+     // Create a variable to store the capitalized name: capCharName
+      const capCharName = object.name
+      // Split the name of the current object by spaces into an array of words using the split method
+      .split(" ")
+      // Capitalize the first character of each word in the capCharName array using the charAt(0) method and toUpperCase() method
+      //  +
+      // Slice the words from the second letter (index of 1) using slice method
+      .map(word => word.charAt(0).toUpperCase() + word.slice(1))
+      // Join the capitalized letters and sliced words back into a single string separated by spaces using join method
+      .join(" ")
+      // Create a sentence using string interpolation with the capCharName variable and the occupation of the current object
+      return `${capCharName} is ${object.occupation}.`
+  })
+}
```

Comment:
This is great! Your inline pseudo code says it all, you don't even need the other pseudo code above